ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -778,45 +729,40 @@ static boolean eligible(BeaconState state, Validator validator) {
    */
   public static void finalUpdates(BeaconState state) throws EpochProcessingException {
     try {
-      UnsignedLong current_epoch = BeaconStateUtil.get_current_epoch(state);
-      UnsignedLong next_epoch = BeaconStateUtil.get_next_epoch(state);
-      UnsignedLong ENTRY_EXIT_DELAY = UnsignedLong.valueOf(Constants.ACTIVATION_EXIT_DELAY);
-      UnsignedLong LATEST_INDEX_ROOTS_LENGTH =
-          UnsignedLong.valueOf(Constants.LATEST_ACTIVE_INDEX_ROOTS_LENGTH);
-      UnsignedLong LATEST_RANDAO_MIXES_LENGTH =
-          UnsignedLong.valueOf(Constants.LATEST_RANDAO_MIXES_LENGTH);
-      UnsignedLong LATEST_PENALIZED_EXIT_LENGTH =
-          UnsignedLong.valueOf(Constants.LATEST_SLASHED_EXIT_LENGTH);
+      long current_epoch = BeaconStateUtil.get_current_epoch(state);
+      long next_epoch = BeaconStateUtil.get_next_epoch(state);
+      long ENTRY_EXIT_DELAY = Constants.ACTIVATION_EXIT_DELAY;
+      long LATEST_INDEX_ROOTS_LENGTH = Constants.LATEST_ACTIVE_INDEX_ROOTS_LENGTH;
+      long LATEST_RANDAO_MIXES_LENGTH = Constants.LATEST_RANDAO_MIXES_LENGTH;
+      long LATEST_PENALIZED_EXIT_LENGTH = Constants.LATEST_SLASHED_EXIT_LENGTH;
 
       // update hash tree root
-      int index = next_epoch.plus(ENTRY_EXIT_DELAY).mod(LATEST_INDEX_ROOTS_LENGTH).intValue();
+      long index = next_epoch + ENTRY_EXIT_DELAY % LATEST_INDEX_ROOTS_LENGTH;
       List<Bytes32> latest_index_roots = state.getLatest_active_index_roots();
       Bytes32 root =
           HashTreeUtil.integerListHashTreeRoot(
               ValidatorsUtil.get_active_validator_indices(
-                  state.getValidator_registry(), next_epoch.plus(ENTRY_EXIT_DELAY)));
-      latest_index_roots.set(index, root);
+                  state.getValidator_registry(), next_epoch + ENTRY_EXIT_DELAY));
+      latest_index_roots.set((int) index, root);
 
       // update latest penalized balances
-      index = next_epoch.mod(LATEST_PENALIZED_EXIT_LENGTH).intValue();
-      List<UnsignedLong> latest_penalized_balances = state.getLatest_slashed_balances();
-      UnsignedLong balance =
-          latest_penalized_balances.get(current_epoch.mod(LATEST_PENALIZED_EXIT_LENGTH).intValue());
-      latest_penalized_balances.set(index, balance);
+      index = next_epoch % LATEST_PENALIZED_EXIT_LENGTH;
+      List<Long> latest_penalized_balances = state.getLatest_slashed_balances();
+      long balance =
+          latest_penalized_balances.get((int) (current_epoch % LATEST_PENALIZED_EXIT_LENGTH));
+      latest_penalized_balances.set((int) index, balance);
```

Comment:
Same comment as above on these getters/setters using a casted int. That's dangerous if the value is > Integer.MAX_VALUE, because it may become unexpectedly negative, and throw an `IndexOutOfBoundsException` at runtime.

Reply:
Noted.